builtin/settingtypes.txt

```diff
@@ -148,6 +148,11 @@ invert_hotbar_mouse_wheel (Hotbar: Invert mouse wheel direction) bool false
 
 [*Touchscreen]
 
+#    Enables the touch screen.
+#    If disabled, touchscreen wouldn't be usable in-game
+#    Changing this setting requires a restart.
+enable_touch (Enable Touchscreen) bool true
```

Comment:
```suggestion
#    Enables touchscreen controls, allowing you to play the
#    game with a touchscreen
#
#    Changing this setting requires a restart.
enable_touch_controls (Enable Touchscreen Controls) bool true
```

Reply:
It shouldn't actually require a restart, though.

Reply:
Yeah I agree

Reply:
What you propose is the optimal solution, thou I think at this moment it's not only the controls which this option enable, but it alters between two modes: mouse X touchscreen. When touchscreen (controls) is enabled, mouse stops working. So far this MR is maximally simple replacement of hard coded compile-only option. It doesn't bring additional improvements regarding to being able use mouse control while using touchscreen.

Reply:
When touchscreen controls are disabled, you can still use the touchscreen to interact with GUIs as it emulates mouse movements

Reply:
Ahh, even if the touchscreen emulates mouse presses. It won't handle gestures. So this setting does effect the gui too

Reply:
> `Enable Touchscreen`

This should be `Enable touchscreen`. Most setting names are in sentence case (there are some exceptions, but we don't have to make it worse).

> `Enables the touch screen.`

I'd write `Enables touchscreen mode.` instead. Also, looking at `settingtypes.txt`, we usually write "touchscreen" rather than "touch screen".

> `If disabled, touchscreen wouldn't be usable in-game`

I'd write something like `Allows you to play the game with a touchscreen.` instead (copied from rubenwardy).

> `Changing this setting requires a restart.`

This is plain wrong, it doesn't.

--------

So to sum up, I suggest this:

```suggestion
#    Enables touchscreen mode, allowing you to play the game with a touchscreen.
enable_touch (Enable touchscreen) bool true
```